### PR TITLE
added support for custom prefixing in redis

### PR DESCRIPTION
**Pull request by TomasLudvik**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Need to run multiple instances of Shopsys Framework application on same machine
|New feature| Yes 
|BC breaks| No 
|Fixes issues| ... 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

**Comment by PetrHeinz** [comment]
This could be implemented backwards-compatible? Default prefix could be set to just `""`, and the possibility of prefixing explained in the docs.

Check related PRs: #651 (prefixing in elasticsearch), #662 (for details about introduced BC break for monorepo usage)

**Comment by TomasLudvik** [comment]
Hello @vitek-rostislav and @PetrHeinz,

thanks for your suggestions. I have removed all BC breaks. I have also removed UPGRADE.md note and added some info into production docs. I have not added any info into `installation-using-docker-application-setup.md` as in development you have Redis installed in the container with rest of your application.

**Comment by PetrHeinz** [comment]
This would imo need a (at least optional) `shopsys/project-base` upgrade note (because code of project-base changed; this is something that we will not avoid) but also an upgrade note for `shopsys/shopsys` (for similar reason as in #662).

**Comment by grossmannmartin** [comment]
Hi,
I made some testing and all cases I've tried,  passed with no problem.

File: UPGRADE.md [hunk 1]
```diff
@@ -66,6 +66,9 @@ There is a list of all the repositories maintained by monorepo, changes in log b
     - new macro `loadMoreButton` is integrated into `@ShopsysShop/Front/Inline/Paginator/paginator.html.twig`, update files based on commit from [`ajaxMoreLoader is updated and generalized`](https://github.com/shopsys/shopsys/pull/579/files)
 - *(optional)* [#645 SVG icons in generated document](https://github.com/shopsys/shopsys/pull/645)
     - to display svg icons collection correctly in grunt generated document for all browsers please add `src/Shopsys/ShopBundle/Resources/views/Grunt/htmlDocumentTemplate.html` file and update `src/Shopsys/ShopBundle/Resources/views/Grunt/gruntfile.js.twig` based on changes in this pull request
+- *(optional)* [#673 added support for custom prefixing in redis](https://github.com/shopsys/shopsys/pull/673)
+    - add default value (e.g. empty string) for `REDIS_PREFIX` env variable to your `app/config/parameters.yml.dist` and to your `docker/php-fpm/Dockerfile`
```
Comment (grossmannmartin) [inline UPGRADE.md]: parameters.yml file also should be mentioned. If someone has own parameters, yml.dist don't help him.

```suggestion
    - add default value (e.g. empty string) for `REDIS_PREFIX` env variable to your `app/config/parameters.yml.dist` , `app/config/parameters.yml` if you already have your parameters and to your `docker/php-fpm/Dockerfile`
```

Comment (vitek-rostislav) [reply UPGRADE.md]: :+1:

**Pull request merged**